fastparquet/compression.py

```diff
@@ -33,50 +33,67 @@ def gzip_decompress_v2(data):
 else:
     def gzip_compress_v3(data, compresslevel=COMPRESSION_LEVEL):
         return gzip.compress(data, compresslevel=compresslevel)
+    def gzip_decompress(data, uncompressed_size):
+        return gzip.decompress(data)
     compressions['GZIP'] = gzip_compress_v3
-    decompressions['GZIP'] = gzip.decompress
+    decompressions['GZIP'] = gzip_decompress
 
 try:
     import snappy
+    def snappy_decompress(data, uncompressed_size):
+        return snappy.decompress(data)
     compressions['SNAPPY'] = snappy.compress
-    decompressions['SNAPPY'] = snappy.decompress
+    decompressions['SNAPPY'] = snappy_decompress
 except ImportError:
     pass
 try:
     import lzo
+    def lzo_decompress(data, uncompressed_size):
+        return lzo.decompress(data)
     compressions['LZO'] = lzo.compress
-    decompressions['LZO'] = lzo.decompress
+    decompressions['LZO'] = lzo_decompress
 except ImportError:
     pass
 try:
     import brotli
+    def brotli_decompress(data, uncompressed_size):
+        return brotli.decompress(data)
     compressions['BROTLI'] = brotli.compress
-    decompressions['BROTLI'] = brotli.decompress
+    decompressions['BROTLI'] = brotli_decompress
 except ImportError:
     pass
 try:
-    import lz4.frame
-    compressions['LZ4'] = lz4.frame.compress
-    decompressions['LZ4'] = lz4.frame.decompress
+    import lz4.block
+    def lz4_compress(data, **kwargs):
+        try:
+            if kwargs['store_size'] == True:
+                raise RuntimeError(
+                    'store_size cannot be True for LZ4 the compressor'
```

Comment:
Could you clarify the wording here, please?

Reply:
I can. I have to admit, I was in half a mind to just silently override that option if the user specified it, rather than raise an exception. Similarly for the `write_content_size` parameter for zstd.

In both cases, this option, when `True` means "store the uncompressed size with the compressed data". We force it to be `False`, because the size is stored outside the compressed block in the metadata, so it would be redundant to store it in the compressed frame header. 

In essence though, we meet the intent of specifying these to be True by storing the size with in the metadata. So, I'd be comfortable for just silently overriding the option the user has specified, rather than raising an exception. On the other hand, I am happy to leave the exception in.

What do you think?

Reply:
I think I agree with you - this is not the typical compression situation, we save the "metadata" separately, so, effectively, the optional parameter is meaningless here.

Reply:
Um. When you say you agree with me, you mean you're happy to drop the exception raising?

Reply:
:)  
I mean that your point is good, and I'd be happy to drop the exceptions altogether, but I think this is a preference call that I'm happy to leave to you.